### ISSUE TITLE
E576: Error while reading ShaDa file: there is an item at position 270498 that must not be there: Missing itemsare for internal uses only

### ISSUE THREAD

**Issue by yuezk** [neovim/neovim#6875]
- v0.2.0
- Vim 8.0 works well.
- macOS Sierra 10.12.5 (16F73)
- iTerm2 v3.0.15
- `$TERM`: screen-256color

### Actual behaviour
When I start neovim by `nvim`, it prints an error message which is `E576: Error while reading ShaDa file: there is an item at position 270498 that must not be there: Missing items are for internal uses only`. But it works well before, I don't known what caused this.

### Expected behaviour

Start without printing any error message.

### Steps to reproduce using `nvim -u NORC`

```
# It has the same problem.
nvim -u NORC
```

### Similar issue

#6652 
#4108

![image](https://user-images.githubusercontent.com/3297602/27000720-f163b290-4dea-11e7-9944-f30f52cf34c8.png)

**Comment by yuezk** [comment]
With `nvim -u NONE -V`, it prints:

```
$ nvim -u NONE -V
Reading ShaDa file "/Users/kevin/.local/share/nvim/shada/main.shada" info marks oldfiles
E576: Error while reading ShaDa file: there is an item at position 270498 that must not be there: Missing items are for internal uses only
Press ENTER or type command to continue^C
```

So I removed `/Users/kevin/.local/share/nvim/shada/main.shada`. It works well. It's weird, anyway, I can use neovim again.

**Comment by nagy135** [comment]
happened to me today, nothing in my flow changed and yet it happened after opening neovim 9355351754th time ..."solution" above works tho

**Comment by nhooyr** [comment]
Can confirm, I'm seeing the same issue. Might be a newly introduced bug? I built from source at fb52790d1c16afdd12e4e484ee1dbce8d2a161b9 and a commit that modifies man.vim but is surely unrelated.

```
$ nvim --version                    
NVIM v0.5.0-866-g339ca766f
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/Users/nhooyr/src/neovim/neovim/build/config -I/Users/nhooyr/src/neovim/neovim/src -I/Users/nhooyr/src/neovim/neovim/.deps/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include -I/usr/local/opt/gettext/include -I/Users/nhooyr/src/neovim/neovim/build/src/nvim/auto -I/Users/nhooyr/src/neovim/neovim/build/include
Compiled by nhooyr@ien.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```

**Comment by Michael-Gallo** [comment]
I solved this issue by deleting all viminfo files in my ~/local/share/vim directory 

``` rm ~/local/share/vim/viminfo*```
local/share is my XDG_DATA_HOME so
```rm $XDG_DATA_HOME/vim/viminfo*```
is probably the more universal fix

**Comment by EmilSodergren** [comment]
I had three files in ~/.local/share/nvim/shada:
- main.shada
- main.shada.tmp.a
- main.shada.tmp.b

I removed all of them and neovim recreated main.shada on first launch. The error disappeared after this.

**Comment by jahsome** [comment]
For any other tortured souls running nvim on windows, the path for the shada may be `$env:LOCALAPPDATA\nvim-data\shada\`

**Comment by optimizasean** [comment]
This has happened to me just now, fixed it same as above by deleting shada folder:

`rm -rf ~/.local/share/nvim/shada` or `rm -rf $HOME/.local/share/nvim/shada` in case your tilde expansion isn't working nice.

Does anyone know what causes this or just unlucky?

**Comment by optimizasean** [comment]
Note: macOS Monterey 12.6 ^ (latest update as of the above post)

**Comment by polm** [comment]
I'm on ArchLinux and still get this pretty frequently after updates, I just have to delete the shada files every time.

**Comment by dundargoc** [comment]
Reopening as this doesn't seem to be solved.

**Comment by hallettj** [comment]
For anyone coming across this thread who removed `~/.local/share/nvim/shada` and still got this error message, I found that I also had shada files in `~/.local/state/nvim/shada`. Removing those files got rid of the error messages.

**Comment by capnahab** [comment]
Still having this issue despite removing ~/.local/share/nvim/shada ,   and placing 

" change default viminfo file locatoin
if !has('nvim') | set viminfo+=n~/.config/vim/viminfo | endif

in my viminfo.

**Comment by ray-x** [comment]
it happened in go.nvim test pipeline  https://github.com/ray-x/go.nvim/actions/runs/3309897836/jobs/5463533673
Only happend for latest but not for 0.7.2.
I set `shada=NONE` to bypass the shada.

**Comment by optimizasean** [comment]
I haven't seen it happen to me since...may just be something file system related or package manager related not cleaning up what it should?

I think an important question should be: `Should nvim handle this through behaviour _____ on fail?`

I am not sure of the purpose of these files or what causes the failures (not backwards compatible with new versions?).
Should the behavior either be delete the file on failure, load without, or delete the files on update/install?
I am not well enough informed on these files to make an assessment/statement for what default should be or how this should be handled.
At least there should be an error message or something describing which file(s) failed and the action the user should/could take or something in the doc for instances like this. This issue is good enough for most people to find it and figure it out, but we could be a little more user friendly, maybe compile things like this in neovim help/error pages or wikis somewhere.

Thoughts?

**Comment by zachliu** [comment]
> For anyone coming across this thread who removed `~/.local/share/nvim/shada` and still got this error message, I found that I also had shada files in `~/.local/state/nvim/shada`. Removing those files got rid of the error messages.

just happened on my `NVIM v0.9.0-dev-484+gb5edea655` but this saves the day

**Comment by igorbrasileiro** [comment]
> I had three files in ~/.local/share/nvim/shada:
> 
> * main.shada
> * main.shada.tmp.a
> * main.shada.tmp.b

I deleted them and solved!

**Comment by devkvlt** [comment]
I get this error even though the directory `~/.local/share/nvim/shada/` doesn't exist on my system (macOS for reference.)

**Comment by GeckoSplinter** [comment]
I had to delete `~/.local/state/nvim/shada/main.shada`

**Comment by eash-d** [comment]
> For anyone coming across this thread who removed `~/.local/share/nvim/shada` and still got this error message, I found that I also had shada files in `~/.local/state/nvim/shada`. Removing those files got rid of the error messages.

It works ! Orz

**Comment by akasmajhi** [comment]
> .local/share/nvim/shada/main.shada
Thanks you @hallettj 
Deleting /.local/state/nvim/shada solution worked.

**Comment by roubaobaozi** [comment]
macOS ventura 13.3.1 (a) and I got this today, deleting 

~/.local/state/nvim/shada fixed it!

Dunno whether this helps anything:

NVIM v0.9.2
Build type: Release
LuaJIT 2.1.0-beta3

**Comment by clason** [comment]
> deleting shada appears to have no negative effects

of course it does; it removes _all_ shared data.

**Comment by zxuliang** [comment]
~/.local/state/nvim/shada  works && thanks

**Comment by phirecc** [comment]
No more comments about deleting `~/.local/state/nvim/shada` please. Yes, it makes neovim start again. No, it's not a fix.
You can think of shada as a database. For some reason, neovim is corrupting the database. Deleting the database is not a reasonable fix and doesn't contribute to this issue tracker.

**Comment by optimizasean** [comment]
> No more comments about deleting `~/.local/state/nvim/shada` please. Yes, it makes neovim start again. No, it's not a fix. You can think of shada as a database. For some reason, neovim is corrupting the database. Deleting the database is not a reasonable fix and doesn't contribute to this issue tracker.

Definitely not a reasonable fix, but it is safe and works every time -> an unfortunate fix.

It is good to see how widespread it is so I would still encourage people to mark OS, plugins, or other data that could be relevant and to keep the issue live and people productive! I wouldn't discourange anyone from posting, especially not from reacting. There isn't comment spam here (way less than a message a month on average) since it doesn't happen to people often enough to annoy anyone subscribed (or people don't message that often).

> This has been an annoyance across multiple OS (macOS 10-present, Arch, Ubuntu, Manjaro, Windows 10, Windows 11). For such a long lived issue, the visibility is critical as it could turn people away from an excellent editor if it just dies for "no reason" and this gets buried deep in the issues bin. I haven't had the issue in probably 3 months, but I am running latest dev on my daily.

## TLDR: *Tested* - don't hesitate to delete the file, affects nothing and gets you back to work!

As far as I can tell, there isn't really any effect on any plugins, commands, or other things I use and do (zero difference to me for usage/plugins/workflows). I deleted it a few more times for fun on different setups just to see if it would cause issues with various common plugins or systems, but I don't really see a difference. Safe to do with the majority of setups across OS and with many of the top plugins as I went on a plugin spree testing things, so don't hold back if you need to delete - no *risk of brick*™ for anything as far as I have looked.

If anyone knows I think it would be helpful for those running niche setups:
- What do we actually lose by deleting this file?
- Are there any plugins dependent on this? 
- Does deleting cause a visible performance hit?
- Are there any known problems related to deleting it?

Was this patched yet? I haven't seen it in a while so maybe only older versions pre-0.9 suffer?

**Comment by phirecc** [comment]
>What do we actually lose by deleting this file?

See `:help shada`:
```
The ShaDa file is used to store:
- The command line history.
- The search string history.
- The input-line history.
- Contents of non-empty registers.
- Marks for several files.
- File marks, pointing to locations in files.
- Last search/substitute pattern (for 'n' and '&').
- The buffer list.
- Global variables.
```

Deleting the shada file thus for example affects plugins like `fzf.vim` which read recently opened files from it.

**Comment by rileyhawk1417** [comment]
> > What do we actually lose by deleting this file?
> 
> See `:help shada`:
> 
> ```
> The ShaDa file is used to store:
> - The command line history.
> - The search string history.
> - The input-line history.
> - Contents of non-empty registers.
> - Marks for several files.
> - File marks, pointing to locations in files.
> - Last search/substitute pattern (for 'n' and '&').
> - The buffer list.
> - Global variables.
> ```
> 
> Deleting the shada file thus for example affects plugins like `fzf.vim` which read recently opened files from it.

I see so its not really affecting anything besides the inconvenience of waiting for the file to populate again with the data specified above right?

**Comment by phirecc** [comment]
>I see so its not really affecting anything besides the inconvenience of waiting for the file to populate again(which happens when you start messing around with a file after opening) with the data specified above right?

Yes in most cases nothing critical will be lost by deleting the file. It's just an inconvenience if you use recently used files in your workflow, for example.

**Comment by Murdock135** [comment]
Does anyone know how to reproduce it?

**Comment by dfenerski** [comment]
Removing the corrupted files did not help.
`:wshada!` did - https://vi.stackexchange.com/questions/43006/error-while-reading-shada-file

**Comment by acemehadi** [comment]
> macOS ventura 13.3.1 (a) and I got this today, deleting
> 
> ~/.local/state/nvim/shada fixed it!
> 
> Dunno whether this helps anything:
> 
> NVIM v0.9.2 Build type: Release LuaJIT 2.1.0-beta3

This work

**Comment by RealYukiSan** [comment]
> > What do we actually lose by deleting this file?
> 
> See `:help shada`:
> 
> ```
> The ShaDa file is used to store:
> - The command line history.
> - The search string history.
> - The input-line history.
> - Contents of non-empty registers.
> - Marks for several files.
> - File marks, pointing to locations in files.
> - Last search/substitute pattern (for 'n' and '&').
> - The buffer list.
> - Global variables.
> ```

out of curiosity, is there a way to print/dump all of those information?
